### PR TITLE
refs #2 Create PieChart DeviceType

### PR DESCRIPTION
**Pull request by dat78787878**
https://github.com/dat78787878/project-training-2/issues/4

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import logo from './logo.svg';
-import './App.css';
-import Home from './containers/Home'
+import PieChart from "./containers/PieChart/PieChart"
+
+
```
Comment (phungMai234) [inline frontend/src/App.js:3]: remove space

File: frontend/src/components/Error.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const Error = () => {
+  return <div className="error">Error</div>;
+};
+
+export default Error;
```
Comment (phungMai234) [inline frontend/src/components/Error.jsx:7]: components/Error/Error.jsx

File: frontend/src/redux/pieChart/saga.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+    const data = yield call(getPieData);
+    yield put(getDataSuccess(data));
+  } catch (e) {
+    yield put(getDataError());
+  }
+}
+
+function* pieChartSaga() {
+  yield takeLatest("GET_PIE_DATA", getPieChart);
```
Comment (phungMai234) [inline frontend/src/redux/pieChart/saga.js:16]: types.GET_PIE_DATA

File: frontend/src/redux/store.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { createStore, applyMiddleware } from "redux";
+import reducer from "./pieChart/reducer";
+import saga from "./pieChart/saga";
+import createSagaMiddleware from "@redux-saga/core";
+
+const sagaMiddleware = createSagaMiddleware();
+const store = createStore(reducer, applyMiddleware(sagaMiddleware));
```
Comment (phungMai234) [inline frontend/src/redux/store.js:7]: use combineReducers => pieChart: pieChartReducer

File: frontend/src/sass/containers/_pieChart.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.pieChart{
+    width: 30%;
+}
```
Comment (phungMai234) [inline frontend/src/sass/containers/_pieChart.scss:3]: move vào containers/Piechart

**Comment by dat78787878** [comment]
image

File: frontend/src/sass/components/_error.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
+.error{
+    color: #fc0404;
+    font-size: 20px;
+    background-color: rgb(234, 0, 255);
+}
```
Comment (phungMai234) [inline frontend/src/sass/components/_error.scss:5]: move vào components/Error
rename style.module.scss

**Review by phungMai234 — COMMENTED** [review]
c k thấy config eslint, prettier đâu 
https://prettier.io/docs/en/configuration.html
https://eslint.org/docs/user-guide/configuring/configuration-files

**Comment by dat78787878** [comment]
image

**Review by phungMai234 — APPROVED** [review]

**Pull request merged**